### PR TITLE
Run unit tests after building docker image and before pushing to the …

### PR DESCRIPTION
**Pull request by rprimet**
…github docker repository

Addresses #289 

Comments welcome, this is a work in progress.

In particular:

- the conda stuff is icky, but we haven't yet gotten rid of conda...
- setting JWT_SECRET is also icky, but we do that elsewhere in CI 
- running the tests from within the container seemed the most 'faithful' choice (v.s. running them in another environment/image just before the build) and also it's guaranteed not to drift w.r.t the base image in the Dockerfile but I'm willing to hear other proposals/options. Thanks!

### CODE REVIEW

File: .github/workflows/docker-package.yml [hunk 1]
```diff
@@ -18,6 +18,8 @@ jobs:
         docker build . --file docker/Dockerfile \
           --tag docker.pkg.github.com/icubam/icubam/icubam-services:${GITHUB_REF##*/} \
           --tag docker.pkg.github.com/icubam/icubam/icubam-services:${{ github.sha }}
+        # sanity check: run unit tests from within the container
+        docker run -e JWT_SECRET=faketest --mount type=bind,source=$PWD/resources,target=/home/icubam/resources,readonly docker.pkg.github.com/icubam/icubam/icubam-services:${GITHUB_REF##*/} "conda init bash && conda activate icubam && pip install pytest && pytest"
```
Comment (rth) [inline .github/workflows/docker-package.yml:22]: Could we install pytest inside Dockerfile?  it cannot hurt for debugging

**Comment by rth** [comment]
Should we first merge https://github.com/icubam/icubam/pull/270 maybe? Unless there are some blockers for it?

**Comment by rprimet** [comment]
@rth this certainly wouldn't hurt, @pgraverdy what's your take on this?

**Comment by rprimet** [comment]
Yep I agree, I'll wait for #270 to land in master and will update this PR afterwards

**Comment by rth** [comment]
@rprimet the other PR is now merged.

**Comment by rth** [comment]
Thanks @rprimet !

**Pull request merged**